prusti-viper/src/encoder/spec_encoder.rs

```diff
@@ -171,6 +172,111 @@ impl<'p, 'v: 'p, 'tcx: 'v> SpecEncoder<'p, 'v, 'tcx> {
+                            let result_name = format!("_{}_forall", vars.args.len() + 2);
+                            let post_conjunct = vir::Expr::forall(
+                                vars.args
+                                    .iter()
+                                    .map(|(arg, arg_ty)| self.encode_forall_arg(*arg, arg_ty, &format!("{}_{}", vars.spec_id, vars.pre_id)))
+                                    .chain(std::iter::once(vir::LocalVar::new(result_name.clone(), vir::Type::Int)))
+                                    .collect(),
+                                vec![], // TODO: encode triggers
+                                vir::Expr::implies(
+                                    encoded_pres,
+                                    vir::Expr::implies(
+                                        vir::Expr::FuncApp(
+                                            sf_post_name,
+                                            (0 .. vars.args.len())
+                                                .map(|i| vir::Expr::Local(
+                                                    vir::LocalVar::new(format!("_{}_forall", i + 2), vir::Type::Int),
+                                                    vir::Position::default()
+                                                ))
+                                                .chain(std::iter::once(
+                                                    vir::Expr::Local(
+                                                        vir::LocalVar::new(result_name.clone(), vir::Type::Int),
+                                                        vir::Position::default()
+                                                    )))
+                                                .collect(),
+                                            (0 ..= vars.args.len())
+                                                .map(|i| vir::LocalVar::new(format!("_{}", i), vir::Type::Int))
+                                                .collect(),
+                                            vir::Type::Bool,
+                                            vir::Position::default()
+                                        ),
+                                        posts.iter()
+                                            .map(|x| self.encode_assertion(x))
+                                            .collect::<Vec<vir::Expr>>()
+                                            .into_iter()
+                                            .conjoin()
+                                    )
+                                )
+                            );
+
+                            vec![pre_conjunct, post_conjunct]
+                                .into_iter()
+                                .conjoin()
+                        }
+                        _ => unreachable!()
+                    }
+                } else {
+                    // TODO
```

Comment:
Could you describe what is left to do?

Reply:
I'm not exactly sure. The condition itself is checking for the result type (which is the closure type) being in the last entry in `typaram_repl`. I assume this *should* be using `self.encoder.current_tymap()` instead, which would probably mean this condition can be dropped.